### PR TITLE
fix: undefined View propTypes issue

### PR DESCRIPTION
**Pull request by capelski**
### Platforms affected
Google Chrome on Windows

### What does this PR do?
Fix propTypes issues on browsers

### What testing has been done on this change?
Launched the app in the Expo Client

### CODE REVIEW

**Comment by justinkchen** [comment]
thanks for this! would be great to get this bumped and merged in! thanks!

**Comment by FINKITT-PTY-LTD** [comment]
Yeah agreed, the merge would make my day guys!

**Comment by jaredswenson** [comment]
This works well, can we get a merge please?

**Comment by JijoBose** [comment]
Thanks, @capelski,  
waiting for this to be merged 🚀

File: src/carousel/Carousel.js [hunk 1]
```diff
@@ -43,8 +43,8 @@ export default class Carousel extends Component {
         autoplayDelay: PropTypes.number,
         autoplayInterval: PropTypes.number,
         callbackOffsetMargin: PropTypes.number,
-        containerCustomStyle: ViewPropTypes ? ViewPropTypes.style : View.propTypes.style,
-        contentContainerCustomStyle: ViewPropTypes ? ViewPropTypes.style : View.propTypes.style,
+        containerCustomStyle: ViewPropTypes ? ViewPropTypes.style : View.propTypes ? View.propTypes.style : () => null,
```
Comment (cyphire) [inline src/carousel/Carousel.js:46]:
```
containerCustomStyle: ViewPropTypes ? ViewPropTypes.style : View.propTypes?.style ? View.propTypes.style : () => null
```
instead... Your's didnt work for me, but the ?. does....

File: src/carousel/Carousel.js [hunk 2]
```diff
@@ -43,8 +43,8 @@ export default class Carousel extends Component {
...
+        contentContainerCustomStyle: ViewPropTypes ? ViewPropTypes.style : View.propTypes ? View.propTypes.style : () => null,
```
Comment (cyphire) [inline src/carousel/Carousel.js:47]: Same as above:

```
        contentContainerCustomStyle: ViewPropTypes ? ViewPropTypes.style : View.propTypes?.style ? View.propTypes.style : () => null,
```

Comment (cyphire) [reply src/carousel/Carousel.js:47]: Sorry - terrible at using this github stuff!!!

You change did not fix the problem, still got red errors in chrome....  The following (just in the Carousel.js) did fix it as a slight mod to yours:

```
containerCustomStyle: ViewPropTypes ? ViewPropTypes.style : View.propTypes?.style ? View.propTypes.style : () => null,
contentContainerCustomStyle: ViewPropTypes ? ViewPropTypes.style : View.propTypes?.style ? View.propTypes.style : () => null,
```

Comment (capelski) [reply src/carousel/Carousel.js:47]: Hi @cyphire. Which device and browser are you testing it from? The fix was working fine on a Windows 10 Chrome browser. The only difference in your code is checking that `View.propTypes` contains a property named `style`. In fact, your code would be more readable if you'd replace the ternary operator with an or operator:

```suggestion
        contentContainerCustomStyle: ViewPropTypes ? ViewPropTypes.style : View.propTypes?.style || () => null,
```

But again, in which case do you get a `View.propTypes` without the `style` property?

**Comment by BlameFelix** [comment]
This works, however I think the underlying problem is [this](https://github.com/necolas/react-native-web/issues/1537#issuecomment-619969653). After applying the fix from the linked comment web also works fine for me.

**Comment by capelski** [comment]
As mentioned in https://github.com/meliorence/react-native-snap-carousel/issues/779#issuecomment-830824204, React Native deprecated `propTypes`. I'm closing this PR since it's not aligned with the React Native evolution and I'm creating https://github.com/meliorence/react-native-snap-carousel/pull/836, which removes `ViewPropTypes` usage instead of mocking it

**Comment by udemezue01** [comment]
How do we solve this, I am getting errors too

File: src/carousel/Carousel.js (hunk 1 above)
Comment (Chris-Imade) [reply src/carousel/Carousel.js:46]: these solutions are awesome I'll try it out right away

**Review by Chris-Imade — COMMENTED** [review]
why isn't this correction merge...when I run yarn add react-native-snap-carousel I still get the not updated files

**Comment by steveberdy** [comment]
If you want this code in your project instead of the current, broken code, you can `npm uninstall react-native-snap-carousel` and then `npm install git+https://github.com/capelski/react-native-snap-carousel` or your own working fork. You don't have to wait for it to be in the official NPM registry.

**Pull request closed without merging**